x/merkledb/codec.go

```diff
@@ -221,18 +220,15 @@ func (*codecImpl) decodeBool(src *bytes.Reader) (bool, error) {
 	}
 }
 
-func (c *codecImpl) encodeInt(dst *bytes.Buffer, value int) {
-	c.encodeInt64(dst, int64(value))
-}
-
-func (*codecImpl) decodeInt(src *bytes.Reader) (int, error) {
+// decodeUint decodes a uvarint from [src] and returns it as an int.
+func (*codecImpl) decodeUint(src *bytes.Reader) (int, error) {
```

Comment:
Arguably this should return a `uint` but a bunch of places where we use this, we want it as an `int`, so idk

Reply:
do we want to just update all of those variables to be uints instead?